tests/test_device_tracker.py

```diff
@@ -0,0 +1,204 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the DeviceTracker and constructor
+"""
+import pytest
+
+import pennylane as qml
+from pennylane import Tracker
+
+
+class TestTrackerCoreBehaviour:
+    def test_default_initialization(self):
+        """Tests default initializalition"""
+
+        tracker = Tracker()
+
+        assert tracker.persistent == False
+        assert tracker.callback is None
+
+        assert tracker.history == dict()
+        assert tracker.totals == dict()
+        assert tracker.latest == dict()
+
+        assert tracker.tracking == False
+
+    def test_device_assignment(self):
+        """Assert gets assigned to device"""
+        dev = qml.device("default.qubit", wires=2)
+
+        tracker = Tracker(dev=dev)
+
+        assert id(dev.tracker) == id(tracker)
+
+    def test_incompatible_device_assignment(self):
+        """Assert exception raised when `supports_tracker` not True"""
```

Comment:
There are two cases here: 1. the `supports_tracker` key is not in the capabilities dictionary (current version) 2. `supports_tracker` is set to `False`. Worth adding 2. too.